CMakeLists.txt

```diff
@@ -13,12 +13,24 @@ include_directories(src/include)
 set(EXTENSION_SOURCES
     src/iceberg_extension.cpp
     src/iceberg_functions.cpp
+    src/catalog_api.cpp
+    src/catalog_utils.cpp 
     src/common/utils.cpp
     src/common/schema.cpp
     src/common/iceberg.cpp
     src/iceberg_functions/iceberg_snapshots.cpp
     src/iceberg_functions/iceberg_scan.cpp
-    src/iceberg_functions/iceberg_metadata.cpp)
+    src/iceberg_functions/iceberg_metadata.cpp
+    src/storage/ic_catalog.cpp
```

Comment:
I feel like this structure could be a bit clearer

What do you think about renaming the IC to IRC (this code is meant for the Iceberg **R**est Catalog after all) and moving everything:
`src/storage/irc/irc_catalog.cpp`?

This would allow us to later on create another catalog which can handle static iceberg tables, similar to https://github.com/duckdb/duckdb-delta/pull/110 which could then live in `src/storage/static/static_catalog.cpp`

Reply:
Do you want all of these files to be `icr_*`or just `ic_catalog.cpp` to be `icr_catalog.cpp`

Reply:
I think `irc_*` (not `icr_` 😄) everywhere that is related to the Iceberg Rest Catalog makes the most sense. We can also look to using C++ namespacing for the iceberg extension at some point, but I think just prefixing everything with IRC is fine for now.